api/service.go

```diff
@@ -0,0 +1,137 @@
+package api
+
+// service.go contains the definition and implementation (business logic) of the
+// login service. Everything here is agnostic to the transport (HTTP).
+
+import (
+	"crypto/sha1"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/microservices-demo/user/db"
+	"github.com/microservices-demo/user/users"
+)
+
+var (
+	passwordSalt    = "passwordsalt"
+	ErrUnauthorized = errors.New("Unauthorized")
+)
+
+// Service is the login service, providing operations for users to login and register.
+type Service interface {
+	Login(username, password string) (users.User, error) // GET /login
+	// Only used for testing at the moment
+	Register(username, password, email string) bool
+	GetUsers(id string) ([]users.User, error)
+	PostUser(u users.User) bool
+	GetAddresses(id string) ([]users.Address, error)
+	PostAddress(u users.Address, userid string) bool
+	GetCards(id string) ([]users.Card, error)
+	PostCard(u users.Card, userid string) bool
+}
+
+// NewFixedService returns a simple implementation of the Service interface,
+func NewFixedService() Service {
+	return &fixedService{}
+}
+
+type fixedService struct{}
+
+func (s *fixedService) Login(username, password string) (users.User, error) {
```

Comment:
The return value doesn't match the spec.
https://github.com/microservices-demo/microservices-demo/blob/master/openapi/specs/users/user.json#L269
Should be array (with one customer?)
Seems like the spec needs fixing, though.